worlds/ahit/DeathWishRules.py

```diff
@@ -0,0 +1,549 @@
+                    add_rule(entrance, lambda state, n=name: get_total_dw_stamps(state, world) >= dw_stamp_costs[n])
+
+        if world.options.DWEnableBonus.value == 0:
+            # place nothing, but let the locations exist still, so we can use them for bonus stamp rules
+            full_clear.address = None
+            full_clear.place_locked_item(HatInTimeItem("Nothing", ItemClassification.filler, None, world.player))
+            full_clear.show_in_spoiler = False
+
+        # No need for rules if excluded - stamps will be auto-granted
+        if world.is_dw_excluded(name):
+            continue
+
+        # Specific Rules
+        modify_dw_rules(world, name)
+
+        main_rule: Callable[[CollectionState], bool]
+        for i in range(len(temp_list)):
+            loc = temp_list[i]
+            data: LocData
+
+            if loc.name == main_objective.name:
+                data = dw_requirements.get(name)
+            else:
+                data = dw_bonus_requirements.get(name)
+
+            if data is None:
+                continue
+
+            if data.hookshot:
+                add_rule(loc, lambda state: can_use_hookshot(state, world))
+
+            for hat in data.required_hats:
+                if hat is not HatType.NONE:
+                    add_rule(loc, lambda state, h=hat: can_use_hat(state, world, h))
+
+            for misc in data.misc_required:
+                add_rule(loc, lambda state, item=misc: state.has(item, world.player))
+
+            if data.paintings > 0 and world.options.ShuffleSubconPaintings.value > 0:
+                add_rule(loc, lambda state, paintings=data.paintings: has_paintings(state, world, paintings))
+
+            if data.hit_type is not HitType.none and world.options.UmbrellaLogic.value > 0:
+                if data.hit_type == HitType.umbrella:
+                    add_rule(loc, lambda state: state.has("Umbrella", world.player))
+
+                elif data.hit_type == HitType.umbrella_or_brewing:
+                    add_rule(loc, lambda state: state.has("Umbrella", world.player)
+                             or can_use_hat(state, world, HatType.BREWING))
+
+                elif data.hit_type == HitType.dweller_bell:
+                    add_rule(loc, lambda state: state.has("Umbrella", world.player)
+                             or can_use_hat(state, world, HatType.BREWING)
+                             or can_use_hat(state, world, HatType.DWELLER))
+
+            main_rule = main_objective.access_rule
+
+            if loc.name == main_objective.name:
+                add_rule(main_stamp, loc.access_rule)
+            elif loc.name == full_clear.name:
+                add_rule(loc, main_rule)
+                # Only set bonus stamp rules if we don't auto complete bonuses
+                if world.options.DWAutoCompleteBonuses.value == 0 \
+                   and not world.is_bonus_excluded(loc.name):
+                    add_rule(bonus_stamps, loc.access_rule)
```

Comment:
If the previous comment is applied, this last block would not go into `add_dw_rules`, but instead get simplified by not needing the if/else at all:
```suggestion
        add_rule(main_stamp, main_objective.access_rule)
        add_rule(full_clear, main_objective.access_rule)
        # Only set bonus stamp rules if we don't auto complete bonuses
        if world.options.DWAutoCompleteBonuses.value == 0 \
           and not world.is_bonus_excluded(full_clear.name):
            add_rule(bonus_stamps, full_clear.access_rule)
```

Reply:
Done